### ISSUE TITLE
How to highlight individual atoms

### ISSUE THREAD

**Issue by luna215** [molstar/molstar#918]
Hi, 

I'm trying to highlight an individual atom from a ligand, but when I do the following:

```typescript
Structure.eachAtomicHierarchyElement(loci?.structure as any, {
    atom: (a) => {
          if (atomIndex === atoms) {
                console.log("HERE");
                const atomLoci = Loci(a.structure, [ {unit: a.unit, indices: [a.element] as any} ])
                // this.canvas.managers.interactivity.lociHighlights.highlightOnly({ atomLoci } as any);
                this.canvas.managers.structure.selection.fromLoci('add', atomLoci);
                this.canvas.managers.camera.focusLoci(atomLoci);
           }
           atoms++;
    }
});
```

the focus part works, but the entire ligand is still selected. (Also, `highlightOnly` doesn't work 😅 ) 

![Screenshot 2023-09-13 at 7 12 52 AM](https://github.com/molstar/molstar/assets/7971405/80adf196-9287-43d7-9c61-45d31b455463)

Is there: 
1. way to only highlight the atom from this approach?
2. Is there a simpler approach then converting an `element-location` to a Loci?

Any help on this will be greatly appreciated 🙏🏽

**Comment by luna215** [comment]
Actually, I got it. I had to change the level for selecting to `element` with this:

```typescript
this.plugin.managers.interactivity.setProps({ granularity: "element" });
```

However, if there is a faster way to select the atom at a given index without having to iterate through all of them, I'd still like to see the solution and see if there's a way to highlight individual atoms with `highOnly` 🤔

**Comment by dsehnal** [comment]
I usually do this by creating a structure selection based on "sourceIndex" property of the atom. It works same way as any other queries. I think it's something like `StructureProperties.atom.sourceIndex` and it has a corresponding mapping in mol-script if you are using that.

**Comment by luna215** [comment]
Hi @dsehnal thanks for the response here. I'm not sure how do the selection with querying but I did manage to do it based on the above code I provided, but if I do it like this and have two ligands in molstar I end up selecting two atoms instead of just one. Could this be a bug? 🤔 
<img width="394" alt="Screenshot 2023-09-18 at 9 29 55 AM" src="https://github.com/molstar/molstar/assets/7971405/6b361c56-906c-4fc4-bbbf-5a715ad2df60">
